### PR TITLE
Add persist and load method for Colbert Index

### PR DESCRIPTION
**Pull request by shanqiu15**
# Description

Introduce utility method for saving and loading the ColBERT index to/from disk

## Type of Change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/examples/managed/ColBERT_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,689 @@
+{
```
Comment (hatianzhang) [inline docs/examples/managed/ColBERT_demo.ipynb:1]: <p>looks there is a special token in the end</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10477/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (hatianzhang) [inline docs/examples/managed/ColBERT_demo.ipynb:1]: <p>could remove such output logs to make the notebook clean</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10477/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (shanqiu15) [reply docs/examples/managed/ColBERT_demo.ipynb:1]: let me remove it.

**Review by hatianzhang — APPROVED** [review]

**Comment by hatianzhang** [comment]
awesome ! thanks for adding those features

**Pull request merged**